### PR TITLE
overlay images on entities support + autotiling on event/no-random rocks and pits + added 2 entities

### PR DESCRIPTION
**Pull request by SnakeBlock**

### CODE REVIEW

**Comment by Zamiell** [comment]
nice work

pr fails black though, can you run black or it, or better yet just set up your IDE to always run black when you save the file

**Review by budjmt — CHANGES_REQUESTED** [review]
Nice work! Just a few suggestions and questions

File: BasementRenovator.py [hunk 1]
```diff
@@ -1474,14 +1485,14 @@ def updateBlockedDoor(self, val, countOnly=False):
...
             for ent in stack:
-                if (
-                    ent.entity.Type == self.entity.Type
-                    and ent.entity.Variant == self.entity.Variant
-                ):
-                    return True
+
+                img = ent.getCurrentImg()
+
+                return img == pitImg
```
Comment (budjmt) [inline BasementRenovator.py:1494]: sorry am dumb
```suggestion
                img = ent.getCurrentImg()

                if img == pitImg:
                    return True
```

**Review by budjmt — CHANGES_REQUESTED** [review]
Sorry one more thing

**Review by budjmt — APPROVED** [review]

**Pull request merged**